test/performance/alignment/test_sequence_generation.hpp

```diff
@@ -0,0 +1,57 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <seqan3/test/performance/sequence_generator.hpp>
+
+inline constexpr size_t default_sequence_length = 150;
+inline constexpr size_t default_sequence_length_variance = 0;
+inline constexpr size_t default_set_size = 500;
+
+template <typename alphabet_t>
+auto generate_data_seqan3(size_t sequence_length = default_sequence_length,
+                          size_t sequence_length_variance = default_sequence_length_variance,
+                          size_t set_size = default_set_size)
+{
+    using sequence_t = decltype(seqan3::test::generate_sequence<alphabet_t>());
+
+    std::vector<sequence_t> vec1;
+    std::vector<sequence_t> vec2;
+
+    for (unsigned i = 0; i < set_size; ++i)
+    {
+        vec1.push_back(seqan3::test::generate_sequence<alphabet_t>(sequence_length,
+                                                                   sequence_length_variance,
+                                                                   i));
```

Comment:
**Attention**: For the random number generator, you have to use the same seed with multiple calls of `std::uniform_int_distribution::operator()` to achieve a uniform distribution. Therefore, you should pass a separate (but for all calls the same) `std::mt19937` object to produce a sequence length. Probably it is then better to produce a length above the function call and just pass the two parameters `len` and `seed` to the `generate_sequence` function.

Reply:
I totally agree with you. This needs to be fixed in general but should be done in a separate PR.